<launch>
    <!-- Path of dataset -->
    <param name="roverMeasureFile" type="string" value="$(find raw_gnss_rtklib)/dataset/UrbanNav/Tokyo/Odaiba/rover_trimble.obs" />
    <param name="baseMeasureFile" type="string" value="$(find raw_gnss_rtklib)/dataset/UrbanNav/Tokyo/Odaiba/base_trim23ble.obs" />
    <param name="BeiDouEmpFile" type="string" value="$(find raw_gnss_rtklib)/dataset/UrbanNav/Tokyo/Odaiba/base.nav" />
    <param name="GPSEmpFile" type="string" value="$(find raw_gnss_rtklib)/dataset/UrbanNav/Tokyo/Odaiba/base.nav" />
    <param name="out_folder" type="string" value="$(find raw_gnss_rtklib)evaluation/g2o_output/gps_lib.pos" />

    <!-- Optimization Parameters -->
    <arg name="verbose" default="false" />
    <arg name="windowSize" default="20" />
    <arg name="maxIterations" default="50" />
    <arg name="maxIterationsEnd" default="10" />
    <arg name="optimizeBiasesAgain" default="true" />
    <arg name="optimizeBiasesAgainEnd" default="true" />
    <arg name="laserInform" default="10" />
    <arg name="filterGPS" default="false" />
    <arg name="maxGPSSpeed" default="20" /> <!-- Limits max distance between GPS poses -->
    <arg name="maxAltToDstPct" default="20" /> <!-- % of altitude change relative to distance -->
    <arg name="decimation" default="1000" /> <!-- n - process each n-th pose -->
    <arg name="posesToProcess" default="12399" /> <!-- limit of poses for processing (before decimation) -->
                                                  <!-- Must not exceed valid poses number in sequence or optimizeAll() will not be called -->
                                                  <!-- Odaiba: 12399 poses-->

    <node name="raw_gnss_rtklib_node" pkg="raw_gnss_rtklib" type="raw_gnss_rtklib_node" output="screen">

        <!--RTKLIB params-->
        <param name="mode" type="int" value="0" />  <!-- GNSS positioning mode, 0: single, 1:DGPS/DGNSS, 2: kinematic -->
        <param name="nf" type="int" value="2" />    <!-- number of frequency (1:L1,2:L1+L2,3:L1+L2+L5) -->
        <param name="soltype" type="int" value="0" />  <!-- 0:forward,1:backward,2:combined -->

        <!-- Optimization parameters -->
        <param name="verbose" type="bool" value="$(arg verbose)" />
        <param name="windowSize" type="int" value="$(arg windowSize)" />
        <param name="maxIterations" type="int" value="$(arg maxIterations)" />
        <param name="maxIterationsEnd" type="int" value="$(arg maxIterationsEnd)" />
        <param name="optimizeBiasesAgain" type="bool" value="$(arg optimizeBiasesAgain)" />
        <param name="optimizeBiasesAgainEnd" type="bool" value="$(arg optimizeBiasesAgainEnd)" />
        <param name="laserInform" type="int" value="$(arg laserInform)" />
        <param name="filterGPS" type="bool" value="$(arg filterGPS)" />
        <param name="maxGPSSpeed" type="int" value="$(arg maxGPSSpeed)" />
        <param name="maxAltToDstPct" type="int" value="$(arg maxAltToDstPct)" />
        <param name="decimation" type="int" value="$(arg decimation)" />
        <param name="posesToProcess" type="int" value="$(arg posesToProcess)" />
    </node>

</launch>